src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -52,20 +53,40 @@ static char* xmlPayloadClean(const char*  payload, const char* payloadWord)
 
 /* ****************************************************************************
 *
+* jsonPayloadClean -
+*/
+static char* jsonPayloadClean(const char* payload)
+{
+  return (char*) strstr(payload, "{");
+}
+
+
+
+/* ****************************************************************************
+*
 * queryForward - 
 *
 * An entity/attribute has been found on some context provider.
 * We need to forward the query request to the context provider, indicated in qcrsP->contextProvider
 *
 * 1. Parse the providing application to extract IP, port and URI-path
-* 2. Render an XML-string of the request we want to forward
+* 2. Render the string of the request we want to forward
 * 3. Send the request to the providing application (and await the response)
-* 4. Parse the XML response and fill in a binary QueryContextResponse
-* 5. Render QueryContextResponse from the providing application
-* 6. 'Fixing' StatusCode
+* 4. Parse the response and fill in a binary QueryContextResponse
+* 5. Fill in the response from the redirection into the response of this function
+* 6. 'Fix' StatusCode
 * 7. Freeing memory
+*
+*
+* FIXME P5: The function 'queryForward' is implemented to pick the format (XML or JSON) based on the
```

Comment:
I think this is related with my previous questions on counters... Not sure of understanding the strategy (Monday effect? :). Maybe we should discuss at skype.